src/adapter/src/coord/peek.rs

```diff
@@ -492,10 +492,16 @@ impl crate::coord::Coordinator {
                 &duration_histogram,
             ) {
                 Ok(rows) => {
+                    let result_size = rows
+                        .clone()
+                        .into_row_iter()
+                        .map(|r| u64::cast_from(r.byte_len()))
+                        .sum();
```

Comment:
I'm not super familiar with `SortedRowCollectionIter`, is cloning and iterating through it expensive? If so, it feels a little bad to do that just to get the result size.